pkg/bloomcompactor/tsdb.go

```diff
@@ -85,7 +88,7 @@ func (b *BloomTSDBStore) LoadTSDB(
 	tenant string,
 	id tsdb.Identifier,
 	bounds v1.FingerprintBounds,
-) (v1.CloseableIterator[*v1.Series], error) {
+) (v1.Iterator[*v1.Series], error) {
```

Comment:
nit: I think you should keep the closable even if you don't use it now you can release the pool later.

Reply:
I'd do it only when we use a pool proviced it's "obvious" we need to release the buffer on the close. But no strong opinion.